Tests/HubTests/HubApiTests.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  HubApiTests.swift
+//
+//  Created by Pedro Cuenca on 20231230.
+//
+
+import XCTest
+@testable import Hub
+
+
+class HubApiTests: XCTestCase {
+
+    override func setUp() {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    // MARK: use a specific revision for these tests
+    
+    func testFilenameRetrieval() async {
+        do {
+            let filenames = try await Hub.getFilenames(from: "coreml-projects/Llama-2-7b-chat-coreml")
+            XCTAssertEqual(filenames.count, 13)
+        } catch {
+            XCTFail("\(error)")
+        }
+    }
+    
+    func testFilenameRetrievalWithGlob() async {
+        do {
+            try await {
+                let filenames = try await Hub.getFilenames(from: "coreml-projects/Llama-2-7b-chat-coreml", matching: "*.json")
```

Comment:
This shows how to retrieve the list of files in a repo. We only get filenames for name, not metadata.